### PR TITLE
Add support for Python 3.6.

### PR DESCRIPTION
**Pull request by icemac**
Add coverage analysis and model .travis.yml alike the one of zope.app.schema.

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -1,27 +1,46 @@
...
+    py34,
+    py35,
+    py36,
+    pypy,
+    pypy3,
+    docs,
+    coverage-report,
 
 [testenv]
+usedevelop = true
```
Comment (mgedmin) [inline tox.ini]: Why is this necessary?

Usually we had a problem with namespace packages, when we used python setup.py test on Python >= 3.5, that would go away with usedevelop = true *or* if we switched to zope-testrunner.  Here I see you do both?

File: tox.ini [hunk 2]
```diff
@@ -1,27 +1,46 @@
...
 commands =
-    python setup.py -q test -q
-# without explicit deps, setup.py test will download a bunch of eggs into $PWD
-# (and it seems I can't use zope.dottedname[testing] here, so forget DRY)
+    coverage run {envbindir}/zope-testrunner --test-path=src {posargs:-vc}
```
Comment (mgedmin) [inline tox.ini]: I slightly prefer `coverage run -m zope.testrunner ...`

File: tox.ini [hunk 3]
```diff
@@ -1,27 +1,46 @@
...
 basepython = python2.7
 commands =
     sphinx-build -ab html -d docs/_build/doctrees docs/ docs/_build/html
-    sphinx-build -ab doctest -d docs/_build/doctrees docs/ docs/_build/doctest
+    coverage run {envbindir}/sphinx-build -ab doctest -d docs/_build/doctrees docs/ docs/_build/doctest
 deps =
     {[testenv]deps}
     Sphinx
+setenv =
+    COVERAGE_FILE=.coverage.{envname}
```
Comment (mgedmin) [inline tox.ini]: Isn't the setenv inherited from the main `[testenv]`?

File: tox.ini (hunk 1 above)
Comment (icemac) [inline tox.ini]: Without this `coverage` has to use the paths where the egg is installed: `.tox/py27/...`, …. Thus it is not able to combine the coverage for the different Python versions.

File: tox.ini (hunk 2 above)
Comment (icemac) [inline tox.ini]: Okay.

File: tox.ini (hunk 3 above)
Comment (icemac) [inline tox.ini]: Good catch.

File: tox.ini (hunk 1 above)
Comment (mgedmin) [inline tox.ini]: Thank you for explaining!

You could also solve that by having a `.coveragerc` with
```
[paths]
source =
    src/zope/authentication
    .tox/*/lib/python*/site-packages/zope/authentication
    .tox/pypy/site-packages/zope/authentication
```
but `usedevelop` is certainly simpler.

Comment (icemac) [inline tox.ini]: @mgedmin You are right, using this configuration is the way which always works.
 I hoped to be able to it an it works in this package. (I checked it again without `usedevelop` and the paths did not get combined.)
But for zopefoundation/zope.securitypolicy#7 `usedevelop` did not help and I had to configure the paths in `.coveragerc`. I did not find any differences in the packages which seem to cause this different behaviour.

I'll leave it now as it is with this reminder for later.
